src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -1724,5 +1722,14 @@ FROM    HDTransactionData
                 yield return addressBase58s;
             }
         }
+
+        /// <inheritdoc />
+        public (IEnumerable<TransactionData>, HdAddress) GetTransactionById(string walletName, uint256 transactionId)
+        {
+            WalletContainer walletContainer = this.GetWalletContainer(walletName);
+            var hdTransactionData = HDTransactionData.GetTransactionById(walletContainer.Conn, transactionId.ToString());
+            var result = hdTransactionData.Select(t => this.ToTransactionData(t, null));
+            return (result, new HdAddress() { Address = hdTransactionData.FirstOrDefault()?.Address, AddressType = hdTransactionData.FirstOrDefault()?.AddressType });
```

Comment:
What is the reasoning behind using `FirstOrDefault` ? What happens if the transaction spends to more than one address in the wallet?

Reply:
@quantumagi I've update the method to incorporate your suggestions.